guides/v2.1/cloud/cdn/fastly-vcl-badreferer.md

```diff
@@ -11,57 +11,104 @@ functional_areas:
+
+1.  Navigate to **Stores** > **Configuration** > **Advanced** > **System** > **Fastly Configuration**.
+
+1.  Expand the **Edge dictionaries** section.
+
+1.  Create the Dictionary container:
+
+    - Click **Add container**.
+
+    -  On the *Container* page, enter a **Dictionary name**. For this example, use the name `referrer_blocklist`.
+
+    -  Select **Activate after the change** to deploy your changes to the service version that you are editing.
+
+    -  Click **Upload** to attach the dictionary to your Fastly service configuration.
+
+1.  Add the list of domain names to block to the the `referrer_blocklist` dictionary:
+
+    -  Click the Settings icon for the `referrer_blocklist` dictonary.
+
+    -  Add and save key-value pairs in the new dictionary. For this example, each **Key** is a website referrer URL that you want to block and the **Value** is `true`. 
+       
+       ![Add bad referrer dictionary items]
+	 
+    -  Click **Cancel** to return to the system configuration page.
 
 For more information on using Edge Dictionaries with your VCL snippets, see Fastly's [Creating and using Edge Dictionaries](https://docs.fastly.com/guides/edge-dictionaries/creating-and-using-dictionaries) and their example [custom VCL snippets](https://docs.fastly.com/guides/edge-dictionaries/creating-and-using-dictionaries#custom-vcl-examples).
 
-## Create badreferer.json {#vcl}
+## Create a custom VCL snippet to block referrer spam
 
-For this example, you may only want to run it against the Production server. You can also add it to Staging for testing.
+The following custom VCL snippet code (JSON format) checks if incoming requests are from a referrer site included in the `referrer_blocklist` edge dictionary. If the site is in the block list, the request is blocked.
 
-Create an `badreferer.json` file with the following JSON content:
 
 ```json
 {
-  "name": "badreferer",
+  "name": "block_bad_referrer",
   "dynamic": "0",
   "type": "recv",
   "priority": "5",
   "content": "set req.http.Referer-Host = regsub(req.http.Referer, \"^https?://?([^:/\\s]+).*$\", \"\\1\"); if (table.lookup(referer_blocklist, req.http.Referer-Host)) { error 403 \"Forbidden\"; }"
 }
 ```
+Review the code to determine if you need to change any values:
+
+  -  `name`: Name for the VCL snippet. For this example, we used the name `block_bad_referrer`.
+  
+  -  `dynamic`: Value 0 indicates that this is a [regular snippet](https://docs.fastly.com/guides/vcl-snippets/using-regular-vcl-snippets)  that you upload to the versioned VCL for the Fastly configuration.
+
+  -  `priority`: Determines when the VCL snippet runs. The priority is set to 5, which means that this snippet code runs before any of the default Magento VCL snippets (magentomodule) that have a priority of 50.
+
+  -  `type`: This VCL is a `recv` snippet type which adds the snippet code to the `vcl_recv`subroutine below the boilerplate VCL and above any objects.
```

Comment:
"Which" seems odd here, as if it needs a comma or needs to be "that" instead. not sure.

Also, the first sentence describes something in the example, but the other list items have a generic definition statement before describing the example usage. Seems like it should describe what `type` is to be consistent. 

Is "boilerplate" used commonly with VCL? If not, I recommend choosing a description easier to translate.

Reply:
Boilerplate is used throughout the Fastly doc, but I agree that it's not a great term. I changed to "default Fastly VCL code"